types/googlemaps/index.d.ts

```diff
@@ -2509,11 +2524,9 @@ declare namespace google.maps {
 
         export class HeatmapLayer extends MVCObject {
             constructor(opts?: HeatmapLayerOptions);
-            getData(): MVCArray;
+            getData<T extends LatLng | WeightedLocation>(): MVCArray<T>;
```

Comment:
The problem if I declare 

```typescript
getData(): MVCArray<LatLng | WeightedLocation>;
```
is that we can not write the following :

```typescript
// Equals property is missing in WeightedLocation
let datas: google.maps.MVCArray<google.maps.LatLng> = heatmap.getData(); 
```
we have to do

```typescript
let datas = heatmap.getData();
(<google.maps.LatLng>datas.getAt(0)).lat();
```

I hesitated but I though it was better to do 
```typescript
let datas = heatmap.getData<google.maps.LatLng>();
datas.getAt(0).lat();
```
with the big drawback that it's a breaking change and can be  legitimately refused by DefinitelyTyped community :)